static/src/javascripts/projects/common/modules/identity/modules/modal.js

```diff
@@ -43,11 +73,24 @@ const getContents = (name: string): Promise<HTMLElement> =>
         })
     );
 
-const show = (name: string): Promise<void> =>
+const show = (
+    name: string,
+    uncheckedNewsletters: ?Array<string>
```

Comment:
this is the main sticking point of my comment above, this makes this code super not reusable. If this were me what I'd do is, from `consent-journey.js`, retrieve and mutate the modal contents with `getContents` (already exported from `modal.js`) before opening the modal